src/serialization/json/JsonSerializationWriter.cs

```diff
@@ -444,7 +472,7 @@ private void WriteNonParsableObjectValue<T>(string? key, T value)
             if(value == null)
                 writer.WriteNullValue();
             else
-                foreach(var oProp in typeof(T).GetProperties())
+                foreach(var oProp in value.GetType().GetProperties())
```

Comment:
why this change?

Reply:
I believe this is covered in the issue #306, but typeof(T) comes back as object and therefore no properties are ever returned by GetProperties(). For a clear demonstration of the issue you can revert this change and run the WriteSampleObjectValue test I modified.

Reply:
Thanks for the additional information. I'd like @andrueastman to provide his input here. I think this should be fine because of the annotation, but would like a second opinion here.

Reply:
Yeah, this change is okay with me really.
The codebase was always using `value.GetType()` but the use of `typeof(T)` snuck in recently in this PR https://github.com/microsoft/kiota-dotnet/pull/291/files when we were trying to iron out AOT/trimming warnings with the test project. So, this is probably looking to fix a regression that we didn't catch.